WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListViewModel.swift

```diff
@@ -20,184 +17,104 @@ final class BlogListViewModel: NSObject, ObservableObject {
+        setupFetchedResultsController()
     }
 
     func searchQueryChanged(_ newText: String) {
         searchText = newText
         updateSearchResults()
     }
 
-    func togglePinnedSite(siteID: NSNumber?) {
-        guard let siteID, let blog = allBlogs.first(where: { $0.dotComID == siteID }) else {
-            return
-        }
-        let isCurrentlyPinned = blog.pinnedDate != nil
-
-        trackPinned(blog: blog)
-
-        blog.pinnedDate = isCurrentlyPinned ? nil : Date()
-
-        contextManager.saveContextAndWait(contextManager.mainContext)
-    }
-
-    func siteSelected(siteID: NSNumber?) {
-        guard let siteID, let blog = allBlogs.first(where: { $0.dotComID == siteID }) else {
-            return
+    func didSelectSite(withSiteID siteID: NSNumber) -> Blog? {
+        guard let blog = rawSites.first(where: { $0.dotComID == siteID }) else {
+            return nil
         }
-
-        trackSiteSelected(blog: blog)
-
+        eventTracker.track(.siteSwitcherSiteTapped, properties: [
+            "section": blog.lastUsed != nil ? "recent" : "all"
+        ])
         blog.lastUsed = Date()
-
         contextManager.saveContextAndWait(contextManager.mainContext)
-    }
-
-    private func trackPinned(blog: Blog) {
-        eventTracker.track(
-            .siteSwitcherPinUpdated,
-            properties: [
-                "blog_id": blog.dotComID ?? "",
-                "pinned": blog.pinnedDate == nil
-            ]
-        )
-    }
-
-    private func trackSiteSelected(blog: Blog) {
-        let sectionName: String
-
-        if blog.pinnedDate != nil {
-            sectionName = "pinned"
-        } else if blog.lastUsed != nil {
-            sectionName = "recent"
-        } else {
-            sectionName = "all"
-        }
-
-        eventTracker.track(
-            .siteSwitcherSiteTapped,
-            properties: [
-                "section": sectionName,
-            ]
-        )
+        return blog
     }
 
     func viewAppeared() {
-        if recentSites.isEmpty && pinnedSites.isEmpty {
+        if recentSites.isEmpty {
             selectedBlog()?.lastUsed = Date()
         }
-
         contextManager.save(contextManager.mainContext)
     }
 
     private func selectedBlog() -> Blog? {
-        return RootViewCoordinator.sharedPresenter.currentOrLastBlog()
+        RootViewCoordinator.sharedPresenter.currentOrLastBlog()
     }
-}
 
-extension BlogListView.Site {
-    init(blog: Blog) {
-        self.init(
-            id: blog.dotComID ?? 0,
-            title: blog.title ?? "",
-            domain: blog.url ?? "",
-            imageURL: blog.hasIcon ? URL(string: blog.icon ?? "") : nil
-        )
+    private func setupFetchedResultsController() {
+        fetchedResultsController.delegate = self
+        do {
+            try fetchedResultsController.performFetch()
+        } catch {
+            wpAssertionFailure("sites-fetch-failed", userInfo: ["error": "\(error)"])
```

Comment:
I realize it's incredibly unlikely that this will happen, but in a `ViewModel` it's pretty easy to have a `var error: Error?` property that surfaces this error to the user as an alert.

WDYT?

Reply:
I can bet money this will never happen 🙅

Reply:
So then should we just `try!`?

I realize it's not "best practice" but if we're that sure I don't think it's worth having the do/try/catch ceremony?